src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -225,11 +225,10 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
 
             var stringComparer = PathUtility.GetStringComparerBasedOnOS();
 
-            var uniqueProjectDependencies = new HashSet<string>(stringComparer);
+            var projects = (await solutionManager.GetNuGetProjectsAsync()).OfType<IDependencyGraphProject>().ToList();
+            var knownProjects = projects.Select(e => e.MSBuildProjectPath).ToHashSet();
```

Comment:
Do we need the string comparer above for the HashSet?

Reply:
Yep, nice catch.